### ISSUE TITLE
CLI: Extensions list enhancements

### ISSUE THREAD

**Issue by Sgitario** [quarkusio/quarkus#18062]
## Description
There are a few drawbacks when using the extension list command: `quarkus extension list`:
1. It behaves differently depending on where you're running the command:
- If current path is a Quarkus application, this command displays the installed Quarkus extensions
- If current path is NOT a Quarkus application, this command displays all the Quarkus extensions

2. Default command `quarkus extension list` behaves as `quarkus extension list --origins`. Example:

```
> quarkus extension list --origins
Jacoco - Code Coverage  999-SNAPSHOT   io.quarkus:quarkus-bom::pom:999-SNAPSHOT
JAXB                                 999-SNAPSHOT   io.quarkus:quarkus-bom::pom:999-SNAPSHOT
```

And I see a couple of problems here: (1) we miss the artifact id that match to the extension name and (2) the version name (aka `999-SNAPSHOT`) is duplicated.

As far I see, the `--concise` option should be the default:

```
> quarkus extension list --concise
Jacoco - Code Coverage  quarkus-jacoco
JAXB                                 quarkus-jaxb
```

And, if we add the `--origins` option, I would have expected to see an additional column:

```
> quarkus extension list --concise
Jacoco - Code Coverage  quarkus-jacoco   io.quarkus:quarkus-bom::pom:999-SNAPSHOT
JAXB                                 quarkus-jaxb       io.quarkus:quarkus-bom::pom:999-SNAPSHOT
```

3. The format of the origin is adding an extra `:`. Example: `io.quarkus:quarkus-bom::pom:999-SNAPSHOT`.
4. The option `--full` have headers and different column order than the rest. Example:

```
extension list --full
Status   Extension                              ArtifactId                 Updated Version          Guide
             Jacoco - Code Coverage      quarkus-jacoco                                           https://quarkus.io/guides/tests-with-coverage
             JAXB                                     quarkus-jaxb                                                                
```

## Implementation Ideas
To sum up, these are the changes I think that are necessary:
- Split the existing command into one for display the installed extensions and another one to list ALL the extensions.
- If no option provided, the `--concise` should be used instead of `--origins`.
- All the options should maintain the column output order:
For `--concise`: `<artifactId> <extensionName>`
For `--origins`: `<artifactId> <extensionName> <origin>`
For `--full`: `<artifactId> <extensionName> <origin> <guide> <status> <updatedVersion>`
- All the options (including `--instalable`) should be sorted. Related issue: https://github.com/quarkusio/quarkus/issues/16154
- Decide whether return headers or not, but do it for all the options.

**Comment by Sgitario** [comment]
cc @maxandersen @ebullient

**Comment by ebullient** [comment]
We will stick with the single command, which includes `--search` and `--installable`, as this is similar to other tools (like `brew`)

In the case of listing from outside of a project, OR listing a specific version, I used `origins` deliberately, to be clear about what version of quarkus extensions were being listed for. It is a very different context to just "list extensions for this project", but isn't different enough to make an entirely different subcommand necessary.

I agree with you re: column ordering.

I think we can/should clean up the origins output (the extra : is due to how the artifact coords print, we should simplify when that segment is empty, which is most of the time).

There is an incoming PR for adding category support: https://github.com/quarkusio/quarkus/pull/17765. Any fix for this will come after that PR is integrated.

**Comment by ebullient** [comment]
I have a partial update (clarifying modes) in #18086 

Sorting out the order that output appears in will come after #17765

**Comment by ebullient** [comment]
#17765 has been merged. extensions are alphabetized and can be filtered by category..
